src/util/mod.rs

```diff
@@ -454,15 +461,86 @@ pub fn is_even(n: usize) -> bool {
     n & 1 == 0
 }
 
+/// Initialize store-dir, create VERSION file, db directory and snap directory.
+pub fn init_store_dir(store_path: &Path) -> io::Result<(PathBuf, PathBuf, PathBuf)> {
+    let mut builder = DirBuilder::new();
+    builder.recursive(true);
+    let version_path = store_path.join(Path::new("VERSION"));
+    let db_path = store_path.join(Path::new("db"));
+    builder.create(&db_path)?;
+    let snap_path = store_path.join(Path::new("snap"));
+    builder.create(&snap_path)?;
+
+    Ok((version_path, db_path, snap_path))
+}
+
+/// Check `TiKV` version.
+/// Note: MUST keep the returned file for holding the lock.
+pub fn check_version(version_path: &Path, db_path: &Path) -> Result<File, String> {
+    let mut version_lock = OpenOptions::new()
+        .write(true)
+        .read(true)
+        .create(true)
+        .open(version_path)
+        .map_err(|e| format!("{:?}", e))?;
+    version_lock.try_lock_exclusive().map_err(|e| {
+        format!(
+            "lock {:?} failed, maybe another instance is using this directory, err {:?}",
+            version_path,
+            e
+        )
+    })?;
+
+    let mut ver = String::new();
+    version_lock
+        .read_to_string(&mut ver)
+        .map_err(|e| format!("{:?}", e))?;
+    if ver.is_empty() {
+        let db_entries = db_path.read_dir().map_err(|e| format!("{:?}", e))?;
+        if db_entries.count() == 0 {
+            version_lock
+                .write_all(super::VERSION.as_bytes())
+                .map_err(|e| format!("{:?}", e))?;
+            Ok(version_lock)
+        } else {
+            Err("miss version".to_owned())
+        }
+    } else {
+        check_version_compatiblity(&ver, super::VERSION)?;
+        Ok(version_lock)
+    }
+}
+
+// It fails when the the major versions are different.
+fn check_version_compatiblity(previous_ver: &str, current_ver: &str) -> Result<(), String> {
+    let previous_ver = Version::parse(previous_ver)
+        .map_err(|e| format!("{:?}", e))?;
+    let current_ver = Version::parse(current_ver).map_err(|e| format!("{:?}", e))?;
+    if previous_ver.major == current_ver.major {
+        Ok(())
+    } else {
+        Err(format!(
```

Comment:
If we follow Semantic Versioning standard, the major version is for incompatibility change after we release 1.0.0.